### ISSUE TITLE
Blobfuse files do not sync after mounting in a container

### ISSUE THREAD

**Issue by robinatmicrosoft** [Azure/azure-storage-fuse#1202]
### Which version of blobfuse was used?
blobfuse2=2.0.4, blobfuse=1.4.1

### Which OS distribution and version are you using?
Ubuntu 20.04.4 LTS

### What was the issue encountered?
Files are sometimes not syncing despite setting --timeout-sec=0 and --direct-io=true. 

### Have you found a mitigation/solution?
Manually dropping the kernel cache with `sysctl -w vm.drop_caches=3` seems to work, but I may not always have write access to /proc/sys file system, so this isn't a long term solution.

**Comment by vibhansa-msft** [comment]
ICM : 401293448
As kernel page cache cleanup works and direct_io does not, it means kernel is not honoring the direct_io flag in this case.
We have observed off late that in containerized mounts this issue is hit. Is this the same case here as well ?

**Comment by souravgupta-msft** [comment]
@robinatmicrosoft, can you please share your mount command and config file used.
Also, to enable direct_io, you will have to pass `-o direct_io` in your mount command instead of `--direct-io=true`.

**Comment by souravgupta-msft** [comment]
The libfuse and file_cache components need correction. The `direct-io` and `timeout-sec` fields are not set correctly. You can refer [this config](https://github.com/Azure/azure-storage-fuse/blob/main/testdata/config/azure_key_directio.yaml) as reference.

```
libfuse:
    attribute-expiration-sec: 0
    entry-expiration-sec: 0
    negative-entry-expiration-sec: 0
    direct-io: true
file_cache:
    path: /scratch/INPUT1_tmp
    max-size-mb: 163840
    timeout-sec: 0
```

Kindly retry after making the above change.

**Comment by robinatmicrosoft** [comment]
@souravgupta-msft The indentation being off was a copy pasting error -- the config was correct (identical to what you shared) when I tried this.

**Comment by vibhansa-msft** [comment]
We are not able to reproduce this locally. If we provide a potential fix will you be able to try it out.

**Comment by robinatmicrosoft** [comment]
Yep, can definitely try it out -- a bit confused though about not being able to reproduce this, since earlier you said you "observed off late that in containerized mounts this issue is hit."?

**Comment by vibhansa-msft** [comment]
Yes we have some customer complaining about it but we ourselves do not have any AKS environment to try this out. It will take some time to build the setup and give it a shot. Meanwhile based on various issues reported in this regard, we have a potential fix.

**Comment by rmacian** [comment]
Beware disabling caches. We have incurred in high costs because Azure charges every list operation of the blob storage.

I left a comment here but I cannot reopen the issue: https://github.com/Azure/azure-storage-fuse/issues/1139

**Comment by vibhansa-msft** [comment]
@rmacian when user disables the caching at various levels in blobfuse, it will result into higher number of calls to storage. Not just list but even get-attribute or file or folder will also hit the storage and all these calls will result into higher billing.
Disabling caching is a conscious choice user has to make considering the cost vs the feature they wish to use.

**Comment by vibhansa-msft** [comment]
All our open issues related to direct_io support have been addressed in 2.1.0 release. Kindly give new binary a try and reopen this report if issue persists.
